freqtrade/tests/test_arguments.py

```diff
@@ -185,3 +185,12 @@ def test_testdata_dl_options() -> None:
     assert args.export == 'export/folder'
     assert args.days == 30
     assert args.exchange == 'binance'
+
+
+def test_check_int_positive() -> None:
+    assert Arguments.check_int_positive(2) == 2
+    assert Arguments.check_int_positive(6) == 6
+    with pytest.raises(argparse.ArgumentTypeError):
+        Arguments.check_int_positive(-6)
+
+    assert Arguments.check_int_positive(2.5) == 2
```

Comment:
I don't remember now, should not it fail here? (2.5 is not an integer)
I think strings are passed to this function from the command line instead of ints or floats...
(Do not remember exactly now, need to sleep...)

Reply:
well noish... - it uses `int(input)` and checks if it's positive.

but yes, it should also work with strings...

Reply:
In the app it's used with a string. And it's intended to be used with the strings. So the passing it a float and obtaining an int instead of an error (as in case of string with float) is not expected behaviour.

Reply:
err no?

https://github.com/freqtrade/freqtrade/blob/1a5dbd29e032d30016ed15ed2aa76e4ba3323ee9/freqtrade/arguments.py#L320-L336

the 2nd argument (--min-trades) defaults to an integer - so i assume it'll pass an int to it (?) (i'll need to debug that when i'm on my real pc ... )

Reply:
I think in this case the check_int_positive() is simply not called and the default value assigned to dest as it is...

Reply:
you could add some nefative values as well for completeness... and zero...

Reply:
that wasn't clear to me ... i added a type-hint now too to be clear that it's passing a string.